### PR TITLE
#2429 Added TypeHandler for BLOB/CLOB of JDBC4.0 implementation

### PR DESCRIPTION
**Pull request by neetapimpalnerkar**
Please review #2429.

### CODE REVIEW

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ Database Access (MyBatis3)
 Overview
 --------------------------------------------------------------------------------
 
-This chapter describes how to access database using \ `MyBatis3 <http://www.mybatis.org/mybatis-3/>`_\ .
+This chapter describes how to access database using \ `MyBatis3 <http://mybatis.org>`_\ .
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 2]
```diff
@@ -748,7 +748,7 @@ MyBatis3 can specify \ ``fetchSize``\  by using 2 methods shown below to control
 
  .. note:: **"Regarding default fetchSize"**
 
-    "Default \ ``fetchSize``\ "can be used in MyBatis3.3.0 and subsequent versions.
+    "Default \ ``fetchSize``\ "can be used in MyBatis3.3.0 and subsequent versions supported in terasoluna-gfw-mybatis3 5.2.0.RELEASE.
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 3]
```diff
@@ -1030,6 +1030,12 @@ Basically, it is used when
 A \ ``TypeHandler`` \  is provided by MyBatis3 for general Java classes like primitive type and primitive wrapper type class.
 Specific settings are not required.
 
+ .. note:: **Regarding implementation for BLOB and CLOB**
+
+    \ ``TypeHandler`` \  added by MyBatis 3.4 implements conversion of BLOB and \ ``java.io.InputStream`` \ , and CLOB and \ ``java.io.Reader`` \  by using API added by JDBC 4.0 (Java 1.6).
+    In case of a JDBC driver with JDBC 4.0 support, it is not necessary to implement a new \ ``TypeHandler`` \  since type handler for conversion of
+    BLOB��\ ``InputStream`` \  and CLOB��\ ``Reader`` \  is enabled by default.
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] This sentence is garbled. Please check this.  I will suggest to change as follow:

"BLOB <-> `InputStream` and CLOB <-> `Reader`"

What do you think ?
If you have more good idea, I'm welcome for your idea !!

Note: Same problem has at other explanation.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 4]
```diff
@@ -3292,8 +3291,8 @@ In the example below, H2 Database is used as the database.
     The links for reference pages of major databases are given below.
 
     * `Oracle 12c <http://docs.oracle.com/database/121/SQLRF/statements_9014.htm>`_
-    * `DB2 11.1 <http://www.ibm.com/support/knowledgecenter/SSEPGG_11.1.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000970.html>`_
-    * `PostgreSQL 9.6 <http://www.postgresql.org/docs/9.6/static/sql-insert.html>`_
+    * `DB2 10.5 <http://www-01.ibm.com/support/knowledgecenter/SSEPGG_10.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000970.html>`_
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 5]
```diff
@@ -3292,8 +3291,8 @@ In the example below, H2 Database is used as the database.
...
+    * `PostgreSQL 9.4 <http://www.postgresql.org/docs/9.4/static/sql-insert.html>`_
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 6]
```diff
@@ -4887,120 +4886,26 @@ Implementation example of mapping file is as given below.
 .. _DataAccessMyBatis3HowToExtendTypeHandler:
 
 
-Implementation of TypeHandler
+Implementatio n of TypeHandler
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 7]
```diff
@@ -5309,7 +5214,7 @@ How to implement a process wherein the search results are downloaded as CSV data
 
     When a query to return a large amount of data is to be described, an appropriate value should be set in \ ``fetchSize``\  attribute.
     \ ``fetchSize``\  is a parameter which specifies data record count to be fetched in a single communication between JDBC driver and database.
-    Note that, "default \ ``fetchSize``\ " can be specified in MyBatis configuration file, in MyBatis3.3.0 and subsequent versions.
+    Note that, "default \ ``fetchSize``\ " can be specified in MyBatis configuration file, in MyBatis3.3.0 and subsequent versions which are supported in terasoluna-gfw-mybatis3 5.2.0.RELEASE.
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 8]
```diff
@@ -5917,8 +5822,8 @@ Based on the application requirement, it is also necessary to check the validity
 In such cases, "a method to execute SQL queued for batch execution" is provided in the Mapper interface."
 
 In MyBatis 3.2, \ ``flushStatements``\  method of \ ``org.apache.ibatis.session.SqlSession``\  interface must be called directly, however,
-a method is supported in MyBatis 3.3.0 and subsequent versions wherein a method which assigns
-\ ``@org.apache.ibatis.annotations.Flush``\  annotation in Mapper interface is created.
+a method is supported in MyBatis 3.3.0 and subsequent versions supported in terasoluna-gfw-mybatis3 5.2.0.RELEASE wherein
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

File: source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst [hunk 9]
```diff
@@ -8070,7 +7975,7 @@ When "Lazy Load" is used, one of the libraries given below is necessary in order
 
 
 CGLIB had been used as default until MyBatis 3.2 series.
-From  MyBatis 3.3.0, JAVASSIST has been used as default at the MyBatis 3.3.0 or later version.
+From  MyBatis 3.3.0, JAVASSIST has been used as default at the MyBatis 3.3.0 or later version, which is supported by terasoluna-gfw-mybatis3 5.2.0.RELEASE.
```
Comment (kazuki43zoo) [inline source_en/ArchitectureInDetail/DataAccessDetail/DataAccessMyBatis3.rst]: - [ ] Why this change is need ? I think this change is not need.

**Comment by kazuki43zoo** [comment]
Hi @neetapimpalnerkar,
Please resolve a build error on Travis CI.
We can't merge until fix a build error and review comments.

**Comment by btratodos** [comment]
Closing this pull request as there are decrepencies in original source file and translated file.
